toolchain/check/pattern_match.cpp

```diff
@@ -280,10 +271,14 @@ auto MatchContext::EmitPatternMatch(Context& context,
     }
     case CARBON_KIND(SemIR::ReturnSlotPattern return_slot_pattern): {
       CARBON_CHECK(kind_ == MatchKind::Callee);
-      return_slot_id_ = context.AddInst<SemIR::ReturnSlot>(
+      auto return_slot_id = context.AddInst<SemIR::ReturnSlot>(
           pattern.loc_id, {.type_id = return_slot_pattern.type_id,
                            .type_inst_id = return_slot_pattern.type_inst_id,
                            .storage_id = entry.scrutinee_id});
+      CARBON_CHECK(
+          !context.scope_stack()
+               .LookupOrAddName(SemIR::NameId::ReturnSlot, return_slot_id)
+               .is_valid());
```

Comment:
I think we should avoid putting code with side-effects in a `CHECK` condition: readers may skip over `CHECK`s when trying to understand the effects of the surrounding code.

Reply:
Yeah, I've been trying to unlearn that habit, but I missed this one. Fixed.